src/lib.rs

```diff
@@ -101,6 +107,130 @@ impl RpcClient for ULinkZenoh {
+        {
+            hashmap_key = format!("{}_{:X}", zenoh_key, rand::random::<u64>());
+        }
+
+        // Setup callback
+        let callback = move |query: Query| {
+            // Create UAttribute
+            let Some(attachment) = query.attachment() else {
+                listener(Err(UStatus::fail_with_code(
+                    UCode::Internal,
+                    "Unable to get attachment",
+                )));
+                return;
+            };
+            let Some(attribute) = attachment.get(&"uattributes".as_bytes()) else {
+                listener(Err(UStatus::fail_with_code(
+                    UCode::Internal,
+                    "Unable to get uattributes",
+                )));
+                return;
+            };
+            let Ok(u_attribute) = Message::decode(&*attribute) else {
+                listener(Err(UStatus::fail_with_code(
+                    UCode::Internal,
+                    "Unable to decode attribute",
+                )));
+                return;
+            };
+            // Create UPayload
+            let u_payload = match query.value() {
+                Some(value) => {
+                    let Ok(encoding) = value.encoding.suffix().parse::<i32>() else {
+                        listener(Err(UStatus::fail_with_code(
+                            UCode::Internal,
+                            "Unable to get payload encoding",
+                        )));
+                        return;
+                    };
+                    UPayload {
+                        length: Some(0),
+                        format: encoding,
+                        data: Some(Data::Value(value.payload.contiguous().to_vec())),
+                    }
+                }
+                None => UPayload {
+                    length: Some(0),
+                    format: UPayloadFormat::UpayloadFormatUnspecified as i32,
+                    data: None,
+                },
+            };
+            // Create UMessage
+            let msg = UMessage {
+                source: Some(method.clone()),
+                attributes: Some(u_attribute),
+                payload: Some(u_payload),
+            };
+            listener(Ok(msg));
+        };
+        if let Ok(queryable) = self
+            .session
+            .declare_queryable(&zenoh_key)
+            .callback_mut(callback)
+            .res()
+            .await
+        {
+            self.queryable_map
+                .lock()
+                .unwrap()
+                .insert(hashmap_key.clone(), queryable);
+        } else {
+            return Err(UStatus::fail_with_code(
+                UCode::Internal,
+                "Unable to register callback with Zenoh",
+            ));
+        }
+
+        Ok(hashmap_key)
+    }
+    async fn unregister_rpc_listener(&self, method: UUri, listener: &str) -> Result<(), UStatus> {
+        // Do the validation
+        if UriValidator::validate(&method).is_err() {
```

Comment:
[Same comment](https://github.com/ZettaScaleLabs/uprotocol-rust-ulink-zenoh/pull/2/files#r1442139810) seems to apply here too.

Reply:
Same as above